index.bs

```diff
@@ -839,6 +838,21 @@ When this method is invoked, the user agent MUST execute the following algorithm
                 1.  Let |attestationObject| be a new {{ArrayBuffer}}, created using |global|'s [=%ArrayBuffer%=], containing the
                     bytes of <code>|credentialCreationData|.[=attestationObjectResult=]</code>'s value.
 
+                1.  Let |attestationPreference| be the value of |options|.{{MakePublicKeyCredentialOptions/attestation}}.
+                    <dl class="switch">
+                    <dt>If the value of |attestationPreference| is "none" (or it has been left unspecified),</dt>
+                    <dd>replace potentially identifying information (such as AAGUID and attestation certificates) 
+                        in the attested credential data and attestation statement with non-identifying versions of the same data.</dd>
+
+                    <dt>If the value of |attestationPreference| is "verifiable",</dt>
+                    <dd>potentially replace the attestation statement with a more privacy-friendly and/or more easily verifiable
+                        version of the same data (for example, by employing a Privacy CA).</dd>
+ 
+                    <dt>If the value of |attestationPreference| is "direct",</dt>
+                    <dd>Obtain consent from the user to relay the authenticator's attestation statement to the RP. If the 
```

Comment:
Consent should not be for normal case.

Reply:
I'd thought we were only talking about obtaining user consent in this case if the authenticator is known to be privacy un-friendly ?

Reply:
The "obtain user consent" handling should be identical for "verifiable" and "direct".  In both cases the authenticator model will be known to some other party (either the Privacy CA or the RP).
In my opinion Authenticator models satisfying the FIDO requirements (i.e. batch size >= 100000 and others) do not lead to a privacy risk and hence do not require yet another user consent (in addition to the general consent of using the authenticator a creating a credential).

Reply:
I don't think we should mandate consent for this mode, either.

It's also semantically weird to require it at this point of the algorithm, as this is already broadly covered in Step 6 of https://w3c.github.io/webauthn/#op-make-cred and Step 5 of https://w3c.github.io/webauthn/#op-get-assertion, though admittedly the user agent only knows for sure how the authenticator responded here in the success branch.

As this is currently written, the user flow might be something like:

1. User clicks "register a strong authenticator"
1. Website renders a page explaining the authentication process and calls `create`
1. User Agent renders a dialog explaining the origin, RP, and the user account asking for the user to consent.  Probably something like, "To continue, touch your flashing token. Otherwise, you can click Cancel or Don't Show This Again."
1. User consents on the token
1. Because of this language, now User Agent says, "This token could be used by $origin to track you. Are you sure?"

I'd much rather roll the consent part together with the first dialog. We know [security indicators are difficult to get right](https://research.google.com/pubs/pub45366.html), and we know that warning fatigue is a real issue. Mandating that forevermore we display a dialog here after the user already interacted with their token seems like a premature UX choice.

If a browser wants to do this, I don't think we need to put a MUST NOT here -- I think we should have the freedom to innovate. Perhaps the first explanatory dialog I mentioned above could remain on screen  after the user expresses consent, and that dialog could be designed to say "Oh hey, since you haven't done this before: this thing you just did, it could be used to track you, so be careful and use another token for sensitive stuff if you need." 

Letting a UA do research and get that right seems great. Making us do it is going to be problematic. (Not even just because of my well-documented issues with mandatory UX in the spec)